### PR TITLE
CONSOLE-2124:  Add badges to operator attributes

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
/assign @TheRealJon
/cc @benjaminapetersen @spadgett 

I am unable to create custom operator for array 'infrastructure' property as a. property of annotations in Operator App Registry. On docker build , I got the below error: 

Screen Shot 2020-03-23 at 11 28 26 AM
Screen Shot 2020-03-23 at 11 28 36 AM

The YAML seems to be valid, also I tried to create same property as a property of spec and it works. 

UPDATE:  issue resolved (@benjaminapetersen )

### CODE REVIEW

**Comment by TheRealJon** [comment]
@cyril, I might be wrong, but shouldn't annotations just be 'key: string' pairs. I don't know if 'key:array' is a valid format for an annotation.

**Comment by cyril-ui-developer** [comment]
> @cyril, I might be wrong, but shouldn't annotations just be 'key: string' pairs. I don't know if 'key:array' is a valid format for an annotation.

**Comment by cyril-ui-developer** [comment]
Quay repo for testing the new annotations:
quay.io/cajieh/web-console:11.0

File: frontend/packages/operator-lifecycle-manager/mocks.ts [hunk 1]
```diff
@@ -737,6 +737,9 @@ export const operatorHubTileViewPageProps = {
       categories: ['messaging', 'streaming'],
       catalogSource: 'testing',
       catalogSourceNamespace: 'openshift-marketplace',
+      paidSubscription: undefined,
```
Comment (benjaminapetersen) [inline frontend/packages/operator-lifecycle-manager/mocks.ts]: Mocks should all be undefined?

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx [hunk 2]
```diff
@@ -142,6 +147,8 @@ export const OperatorHubItemDetails: React.SFC<OperatorHubItemDetailsProps> = ({
               />
               <PropertyItem label="Provider Type" value={providerType || notAvailable} />
               <PropertyItem label="Provider" value={provider || notAvailable} />
+              <PropertyItem label="Infrastructure Features" value={mappedInfraFeatures} />
+              <PropertyItem label="Paid Subscription Required" value={license || notAvailable} />
```
Comment (benjaminapetersen) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx]: License is mapped to Paid Subscription here?

File: frontend/packages/operator-lifecycle-manager/mocks.ts [hunk 3]
```diff
@@ -737,6 +737,9 @@ export const operatorHubTileViewPageProps = {
...
+      license: undefined,
```
Comment (benjaminapetersen) [inline frontend/packages/operator-lifecycle-manager/mocks.ts]: I believe we decided against `license` as a term.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx [hunk 4]
```diff
@@ -248,6 +296,19 @@ const determineAvailableFilters = (initialFilters, items, filterGroups) => {
           });
         }
       }
+
+      if (value !== undefined && Array.isArray(value)) {
```
Comment (benjaminapetersen) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx]: Is `value !== undefined` needed? `Array.isArray(undefined)` is still `false`.

Comment (cyril-ui-developer) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx]: Gotcha

File: frontend/packages/operator-lifecycle-manager/mocks.ts (hunk 1 above)
Comment (cyril-ui-developer) [reply frontend/packages/operator-lifecycle-manager/mocks.ts]: > Mocks should all be undefined?

The values are not used in the unit test, just the shape of the object. I can give them values but it makes no diff.

Comment (benjaminapetersen) [reply frontend/packages/operator-lifecycle-manager/mocks.ts]: ok great. focused for unit tests is good.

**Comment by cyril-ui-developer** [comment]
Steps and format to add badges to operator attributes
Add `operators.openshift.io/infrastructure-features` and `operators.openshift.io/valid-subscription` annotation to operator CSV

Format 1:
annotation values are in String, and these two annotation values are in Array

apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
metadata:
  annotations:
    operators.openshift.io/infrastructure-features: '["Disconnected", "Proxy", "FIPS Mode"]'
    operators.openshift.io/valid-subscription: '["3Scale Commercial License", "Red Hat Managed Integration"]'

Format 2:
Use |- to interpret Array

apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
metadata:
  name: keda.v1.3.0
  namespace: placeholder
  annotations:
    operators.openshift.io/infrastructure-features: |-
      ["Proxy", "FIPS Mode"]
    operators.openshift.io/valid-subscription: |-
      ["Red Hat Managed Integration"]

Screenshots attached:
Screen Shot 2020-04-20 at 4 31 26 PM
Screen Shot 2020-04-20 at 4 32 14 PM

**Pull request merged**